contracts/src/v0.8/dev/automation/upkeeps/MercuryRegistry.sol

```diff
@@ -0,0 +1,280 @@
+    s_deviationPercentagePPM = deviationPercentagePPM;
+    s_stalenessSeconds = stalenessSeconds;
+
+    // Store desired feeds.
+    setFeeds(feedIds, feedNames);
+  }
+
+  // Returns a user-defined batch of feed data, based on the on-chain state.
+  function getLatestFeedData(string[] memory feedIds) external view returns (Feed[] memory) {
+    Feed[] memory feeds = new Feed[](feedIds.length);
+    for (uint256 i = 0; i < feedIds.length; i++) {
+      feeds[i] = s_feedMapping[feedIds[i]];
+    }
+
+    return feeds;
+  }
+
+  // Invoke a feed lookup through the checkUpkeep function. Expected to run on a chron schedule.
+  function checkUpkeep(bytes calldata /* data */) external view override returns (bool, bytes memory) {
+    string[] memory feeds = s_feeds;
+    return revertForFeedLookup(feeds);
+  }
+
+  // Extracted from `checkUpkeep` for batching purposes.
+  function revertForFeedLookup(string[] memory feeds) public view returns (bool, bytes memory) {
+    uint256 blockNumber = ChainSpecificUtil.getBlockNumber();
+    revert FeedLookup(c_feedParamKey, feeds, c_timeParamKey, blockNumber, "");
+  }
+
+  // Filter for feeds that have deviated sufficiently from their respective on-chain values, or where
+  // the on-chain values are sufficiently stale.
+  function checkCallback(
+    bytes[] memory values,
+    bytes memory lookupData
+  ) external view override returns (bool, bytes memory) {
+    bytes[] memory filteredValues = new bytes[](values.length);
+    uint256 count = 0;
+    for (uint256 i = 0; i < values.length; i++) {
+      Report memory report = getReport(values[i]);
+      string memory feedId = bytes32ToHexString(abi.encodePacked(report.feedId));
+      Feed memory feed = s_feedMapping[feedId];
+      if (
+        (report.observationsTimestamp - feed.observationsTimestamp > s_stalenessSeconds) ||
+        deviationExceedsThreshold(feed.price, report.price)
+      ) {
+        filteredValues[count] = values[i];
+        count++;
+      }
+    }
+
+    // Adjusts the length of the filteredValues array to `count` such that it
+    // does not have extra empty slots, in case some items were filtered.
+    assembly {
+      mstore(filteredValues, count)
+    }
+
+    bytes memory performData = abi.encode(filteredValues, lookupData);
+    return (filteredValues.length > 0, performData);
+  }
+
+  // Use deviated off-chain values to update on-chain state.
+  function performUpkeep(bytes calldata performData) external override {
+    (bytes[] memory values /* bytes memory lookupData */, ) = abi.decode(performData, (bytes[], bytes));
+    for (uint256 i = 0; i < values.length; i++) {
+      // Verify and decode the Mercury report.
+      Report memory report = abi.decode(s_verifier.verify(values[i]), (Report));
+      string memory feedId = bytes32ToHexString(abi.encodePacked(report.feedId));
+
+      // Feeds that have been removed between checkUpkeep and performUpkeep should not be updated.
```

Comment:
I guess this is fine since on the next call to `checkUpkeep` those disabled feeds won't be included?

Reply:
Yeah the off-chain node should just see this simulation revert and then not post a transaction. Worst case timing-wise we have a single on-chain revert, and then things go back to normal.